isis/src/control/apps/findfeatures/FastGeom.cpp

```diff
@@ -62,24 +67,398 @@ FastGeom::~FastGeom() { }
+  QString fg_algorithm = m_parameters.get("FastGeomAlgorithm", "Radial").toLower();
+  bool radial_method = ( "grid" != fg_algorithm );
+  logit <<   "  FastGeomAlgorithm:       " << fg_algorithm << "\n";
+  logit <<   "  FastGeomPoints:          " << m_fastpts << "\n";
+  logit <<   "  FastGeomTolerance:       " << m_tolerance << "\n";
+  logit <<   "  FastGeomSampleTolerance: " << fg_sample_tolerance << "\n";
+  logit <<   "  FastGeomLineTolerance:   " << fg_line_tolerance << "\n\n";
+
+  // Run the requested algorithm
+  if ( true == radial_method) {
+    logit << "--> Using Radial Algorithm train-to-query mapping <--\n";
+
+    // Compute maximum radial track at the center of the image to
+    // the corner and scaling parameters
+    double fg_max_radius    = std::sqrt( (q_samps * q_samps) + ( q_lines * q_lines) );
+    double fg_radius_factor = toDouble(m_parameters.get("FastGeomRadiusFactor", "0.035"));
+    double fg_point_factor  = toDouble(m_parameters.get("FastGeomPointFactor", "0.10"));
+
+    // No check for explict specs
+    // Allow user explicit radius delta increments - minimum is the length of the
+    // diagonal of 1 pixel ( 2 * sqrt(1^2 + 1^2))
+    double fg_radius_inc = qMax( 2.8284, (fg_max_radius * fg_radius_factor) );
+    if ( m_parameters.exists("FastGeomRadiusIncrement") ) {
+      fg_radius_inc = toDouble(m_parameters.get("FastGeomRadiusIncrement"));
+    }
+
+    // Allow user explicit point/radius increments. A 0 increment would only
+    // get a single point at all radii at the end of the reference vector.
+    // Probably bad, but we will allow it as an extreme.
+    double fg_point_inc = qMax( 0.0, (fg_radius_inc * fg_point_factor) );
+    if ( m_parameters.exists("FastGeomPointIncrement") ) {
+      fg_point_inc = toDouble(m_parameters.get("FastGeomPointIncrement"));
+    }
+
+    // Lets report what we got
+    logit <<   "  FGMaximumRadius:         " << fg_max_radius << "\n";
+    logit <<   "  FGRadiusFactor:          " << fg_radius_factor << "\n";
+    logit <<   "  FGPointFactor:           " << fg_point_factor << "\n";
+    logit <<   "  FGRadiusIncrement:       " << fg_radius_inc << "\n";
+    logit <<   "  FGPointIncrement:        " << fg_point_inc << "\n";
+    logit <<   "  TotalRadialTracks:       " << fg_max_radius / fg_radius_inc << "\n";
+
+    logit.flush();
+
+    // Rounding constant
+    const double FG_Epsilon(0.0001);
+
+    // Unit vector on circle points east
+    double v_x0 = 1.0;
+    double v_y0 = 0.0;
+
+    // Center of query image
+    double c_x = q_samps / 2.0;
+    double c_y = q_lines / 2.0;
+
+    // The Circle Loop - compute points along a circle at radial
+    // distance (pixels) from center
+    double points_per_radius = 0.0; // Gets the center point on 1st iteration
+    double fg_radius = 0.0;
+    while ( fg_radius <= fg_max_radius ) {
+
+      // Compute the number points / radial
+      double r_points = qMax(1.0, points_per_radius);
```

Comment:
That is an implementation choice. This is something I can test but I think it would produce strange patterns due to other floating point elements of the algorithm.